demo/android/Activity/koala-activity-demo-app/src/main/java/ai/picovoice/koalaactivitydemo/MainActivity.java

```diff
@@ -0,0 +1,379 @@
+
+            started.set(true);
+
+            Executors.newSingleThreadExecutor().submit((Callable<Void>) () -> {
+                Process.setThreadPriority(Process.THREAD_PRIORITY_URGENT_AUDIO);
+                read();
+                return null;
+            });
+        }
+
+        void stop() throws InterruptedException, IOException {
+            if (!started.get()) {
+                return;
+            }
+
+            stop.set(true);
+
+            while (!stopped.get()) {
+                Thread.sleep(10);
+            }
+
+            writeWavHeader(referenceFile, (short) 1, (short) 16, 16000, totalSamplesWritten);
+            writeWavHeader(enhancedFile, (short) 1, (short) 16, 16000, totalSamplesWritten);
+            referenceFile.close();
+            enhancedFile.close();
+
+            started.set(false);
+            stop.set(false);
+            stopped.set(false);
+        }
+
+        @SuppressLint({"MissingPermission", "SetTextI18n"})
+        private void read() throws KoalaException {
+            final int minBufferSize = AudioRecord.getMinBufferSize(
+                    koala.getSampleRate(),
+                    AudioFormat.CHANNEL_IN_MONO,
+                    AudioFormat.ENCODING_PCM_16BIT);
+            final int bufferSize = Math.max(koala.getSampleRate() / 2, minBufferSize);
+
+            AudioRecord audioRecord = null;
+
+            short[] frameBuffer = new short[koala.getFrameLength()];
+
+            try {
+                audioRecord = new AudioRecord(
+                        MediaRecorder.AudioSource.MIC,
+                        koala.getSampleRate(),
+                        AudioFormat.CHANNEL_IN_MONO,
+                        AudioFormat.ENCODING_PCM_16BIT,
+                        bufferSize);
+                audioRecord.startRecording();
+
+                final int koalaDelay = koala.getDelaySample();
+
+                totalSamplesWritten = 0;
+                int enhancedSamplesWritten = 0;
+                while (!stop.get()) {
+                    if (audioRecord.read(frameBuffer, 0, frameBuffer.length) == frameBuffer.length) {
+                        final short[] frameBufferEnhanced = koala.process(frameBuffer);
+
+                        writeFrame(referenceFile, frameBuffer);
+                        totalSamplesWritten += frameBuffer.length;
+                        if (totalSamplesWritten >= koalaDelay) {
+                            writeFrame(enhancedFile, frameBufferEnhanced);
+                            enhancedSamplesWritten += frameBufferEnhanced.length;
+                        }
+                    }
+
+                    if ((totalSamplesWritten / koala.getFrameLength()) % 10 == 0) {
+                        runOnUiThread(() -> {
+                            int secondsRecorded = (totalSamplesWritten / koala.getSampleRate());
+                            recordedText.setText("Recorded: " + secondsRecorded + "s");
```

Comment:
can we do a float to one decimal place for recording time? Looks more familiar as a recording timer with some milliseconds